### PR TITLE
Issue #11014 - Improve Relative Redirect Handling in Jetty 10/11

### PR DESCRIPTION
**Pull request by gregw**
#11014
+ Common relative redirect method.
+ Use the relative redirect method from redirection rules.

### CODE REVIEW

File: jetty-server/src/main/java/org/eclipse/jetty/server/Response.java [hunk 1]
```diff
@@ -622,6 +595,48 @@ public void sendRedirect(String location) throws IOException
...
 
+    /**
+     * Common point to generate a proper "Location" header for redirects.
+     *
+     * @param request the request the redirect should be based on (needed when relative locations are provided, so that
+     * server name, scheme, port can be built out properly)
+     * @param location the location URL to redirect to (can be a relative path)
+     * @return the full redirect "Location" URL (including scheme, host, port, path, etc...)
+     */
+    public static String toRedirectURL(final HttpServletRequest request, String location)
```
Comment (sbordet) [inline jetty-server/src/main/java/org/eclipse/jetty/server/Response.java]: Rename to `toRedirectURI()`, as RFC and Jetty 12 use "URI", not "URL".

Drop `final` for the first parameter.

Also, should clarify whether the `location` is already encoded or not?
E.g. `location: newPath?p=/oldPath` should the `/` be already encoded?

Perhaps it's just enough to change the variable name to `encodedLocation` to make it clear, but callers need to know.

**Review by sbordet — CHANGES_REQUESTED** [review]

**Review by lorban — APPROVED** [review]

File: jetty-server/src/main/java/org/eclipse/jetty/server/Response.java [hunk 2]
```diff
@@ -622,6 +595,49 @@ public void sendRedirect(String location) throws IOException
...
+    /**
+     * Common point to generate a proper "Location" header for redirects.
+     *
+     * @param request the request the redirect should be based on (needed when relative locations are provided, so that
+     * server name, scheme, port can be built out properly)
+     * @param location the location as an absolute URI or an encoded relative path. A relative path starting with '/'
+     *                 is relative to the root, otherwise it is relative to the request path.
+     * @return the full redirect "Location" URL (including scheme, host, port, path, etc...)
+     */
+    public static String toRedirectURI(final HttpServletRequest request, String location)
```
Comment (sbordet) [inline jetty-server/src/main/java/org/eclipse/jetty/server/Response.java:607]: Drop `final`.

**Review by sbordet — APPROVED** [review]
Just a nit on the use of `final`.

**Review by joakime — APPROVED** [review]

File: jetty-server/src/main/java/org/eclipse/jetty/server/Response.java (hunk 2 above)
Comment (gregw) [reply jetty-server/src/main/java/org/eclipse/jetty/server/Response.java:607]: Ooops forgot.   It was there in the original method and gone in 12.  I'll keep here to not lose the green ticks.

**Pull request merged**